### PR TITLE
Fixes AT gun APHE chain explosions

### PR DESCRIPTION
**Pull request by DeltaFire15**
## About The Pull Request
Basically, the way explosive (rocket) ammo works is that on_turf_hit, it explodes, which generally works all fine since a rocket doesn't tend to pass through walls. Hoowever, when there is say, a shell that passes through walls and is explosive, said code causes issues, namely exploding on every closed turf (mainly walls) and stacking explosions, amping its lethality to questionable levels. ~~This resolves this problem via only calling on_turf_hit()'s drop_nade() if said turf is the original intended target, as that is the only situation where it *should* explode on hitting a turf (since hitting xenos or whatever else is handled in the other on_hit procs and it doesn't pass through mobs / simillar anymore)~~

Instead of what I did before, I instead resorted to having ammo not do on_hit_turf except for the final (clicked target) turf if it has both explosive and pass_turf as flags, due to the previous implementation still causing dual explosions on the final turf. (I.. guess I could turn it into a seperate flag if wanted? It's just too oddly specific to really be worth making it its own bitflag which is why I did it this way)
Feel free to suggest alternatives if you can think of a cleaner way of doing this, or if you do want me to turn it into a flag.

## Why It's Good For The Game
Fix man good.

## Changelog
:cl:
fix: APHE only explodes when it hits something that stops its movement, not when passing through walls on its way there.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/projectile.dm [hunk 1]
```diff
@@ -569,7 +569,8 @@ So if we are on the 32th absolute pixel coordinate we are on tile 1, but if we a
 
 /obj/projectile/proc/scan_a_turf(turf/turf_to_scan, cardinal_move)
 	if(turf_to_scan.density) //Handle wall hit.
-		ammo.on_hit_turf(turf_to_scan, src)
+		if((ammo.flags_ammo_behavior & (AMMO_EXPLOSIVE|AMMO_PASS_THROUGH_TURF)) != (AMMO_EXPLOSIVE|AMMO_PASS_THROUGH_TURF))
```
[the remark this answers is not in the document]
Comment (DeltaFire15) [reply code/modules/projectiles/projectile.dm:572]: Well, the & operator isn't exactly "has a flag". In case of checking for a singular flag, it works fine as such. However, lets say you check for a combined bitflag of (1<<0)|(1<<1) which translates to 3. If you check it for something like 0 it returns false and if you check 3 it returns true. Seems okay right? Well, lets check something that only has one of the two, e.g. (1<<1), or well, a 2. That also returns true. Why? Well, (1<<1) & (1<<0)|(1<<1) is 2, which by byond statement conditions is true. So we need to check for if both the bitfields are actually present, which in this case would be [input] & (1<<0)|(1<<1) == (1<<0)|(1<<1), aka, the returned value has to be 3, not 2 or 1, which it might return and let through otherwise.

In this case it's a != because anything except those with both flags are supposed to pass the check.

tl;dr checking for multiple bitflags in one check needs a == like this one to avoid false positives a simple & could otherwise cause.
